recipe/meta.yaml

```
@@ -1,208 +1,211 @@
-    sha256: 485968734706eeffcd3adc3b2d2086e59be7ff3ddd907e96f1eb97335beb344a  # [linux64 and (cuda_compiler_version or "").startswith("12")]
+  {% set target_path = os.sep.join(["", "targets", "%s-%s" % (march, kname)]) %}
+  {% set devzone_base = "https://developer.download.nvidia.com/compute" %}
+  {% set devzone = "%s/%s/redist/%s/%s-%s" % (devzone_base, package, package, kname, march) %}
+  
+  {% set devzone = "%s/%s-%s-%s-%s_cuda%s-archive.tar.xz" % (devzone, package, kname, march, sdk_version_string, cuda_major) %}
+
+  {% set github_base = "https://github.com/nvidia" %}
+  {% set github = "%s/%s" % (github_base, package) %}
+  {% set github = "%s/archive/refs/tags/v%s.tar.gz" % (github, ".".join(sdk_version[:3])) %}
+{% endif %}
 
-  - url: https://developer.download.nvidia.com/compute/cuquantum/redist/cuquantum/linux-ppc64le/cuquantum-linux-ppc64le-{{ version }}_cuda{{ cuda_major }}-archive.tar.xz  # [ppc64le]
-    sha256: 8767174f86211ca3610d91bdf8c6b8145408836a3d14808414297aef3631fd43  # [ppc64le and (cuda_compiler_version or "").startswith("11")]
-    sha256: 06d63f5c3a7a3f3e910259c1bfca3e7fdab72d0378099002941bc0a743f35694  # [ppc64le and (cuda_compiler_version or "").startswith("12")]
+package:
+  name: cuquantum-sdk
+  version: {{ ".".join(sdk_version[:3]) }}
 
-  - url: https://developer.download.nvidia.com/compute/cuquantum/redist/cuquantum/linux-sbsa/cuquantum-linux-sbsa-{{ version }}_cuda{{ cuda_major }}-archive.tar.xz  # [aarch64]
-    sha256: 4b695e25f8e2bfdaa04f68c379fe2001b6fdb8fe63169780b584f7b49acd0312  # [aarch64 and (cuda_compiler_version or "").startswith("11")]
-    sha256: 9ca173c8b58ac5c097a8c69f4900bc9ef0179def721804f1e9f033677445cafc  # [aarch64 and (cuda_compiler_version or "").startswith("12")]
+source:
+  - url: {{ devzone }}  # [cuda_compiler_version in ("11.8", "12.0", "12.6") and (linux64 or aarch64)]
+    sha256: a36123e52daeae5e6e5c543b12d45fb41411280ceb924bbde54d4d1051cf10ef  # [cuda_compiler_version == "11.8" and linux64]
+    sha256: 4640da62e66f17671c8d89be93a439b8a7a13169f7aa902a3e9b6cf676d4bffd  # [cuda_compiler_version in ("12.0", "12.6") and linux64]
+    sha256: 16c1aba50c657456928386a8597d4b766daa3c6f69da738a9568ad4ac97094ae  # [cuda_compiler_version == "11.8" and aarch64]
+    sha256: e3e4a289887d17d4b78eaac97f2fa03b2de9d832f918eb48ab97de997c104ce6  # [cuda_compiler_version in ("12.0", "12.6") and aarch64]
 
-  - url: https://github.com/NVIDIA/cuQuantum/archive/refs/tags/v{{ ".".join(version.split(".")[:3]) }}.tar.gz
-    sha256: 6e4a5ad168e8aaa422f338c2a273242d3c9aa22be081a4065f85d75b6eecda7a
+  - url: {{ github }}  # [linux64 or aarch64]
+    sha256: a4b0965e7193515930542cc558d922e24c1732d8240efde3e622e84d6c4416fe  # [linux64 or aarch64]
 
 build:
   number: {{ build_num }}
-  skip: true  # [not linux or cuda_compiler_version not in ("11.8", "12.0")]
+  skip: true  # [not cuda_compiler_version in ("11.8", "12.0", "12.6")]
```

```suggestion
  skip: true  # [cuda_compiler_version in ("None", None)]
```
I suggesting skipping on whether cuda is None instead because otherwise the recipe will not automatically build when the channel migrates to the next CUDA 12.x release.

---

I am now skipping on cuda major version not being 11 or 12, this should automatically build for future 12.x releases, right?